engine/app.py

```diff
@@ -18,9 +18,10 @@ async def main(websocket):
         endMove = moves[2]
 
         # Do move and update board
-        board.move_piece(startMove, endMove)
-
-        # Send resulting board state
-        await websocket.send(board.board_to_json())
+        if(board.can_move_piece(startMove, endMove)):
+            board.move_piece(startMove, endMove)
+            
+            # Send resulting board/game state
+            await websocket.send(board.game_to_json())            
```

Comment:
This is good for now, later we will add some logic to not send the entire state of the game on every update,